File: TwoCaptcha/Captcha/GeeTestV4.cs
namespace TwoCaptcha.Captcha
{
    public class GeeTestV4 : Captcha
    {
        public GeeTestV4() : base()
        {
            parameters["method"] = "geetest_v4";
        }

        public void SetCaptchaId(string captchaId)
        {
            parameters["captcha_id"] = captchaId;
        }

        public void SetChallenge(string challenge)
        {
            parameters["challenge"] = challenge;
        }

        public void SetUrl(string url)
        {
            parameters["pageurl"] = url;
        }

        public void SetApiServer(string apiServer)
        {
            parameters["api_server"] = apiServer;
        }
    }
}

File: TwoCaptcha/Captcha/GeeTestV4.java
package com.twocaptcha.captcha;

public class GeeTestV4 extends Captcha {

    public GeeTestV4() {
        super();
        params.put("method", "geetest_v4");
    }

    public void setChallenge(String challenge) {
        params.put("challenge", challenge);
    }

    public void setUrl(String url) {
        params.put("pageurl", url);
    }

    public void setApiServer(String apiServer) {
        params.put("api_server", apiServer);
    }

    public void setCaptchaId(String captchaId) {
        params.put("captcha_id", captchaId);
    }    

}
